cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/ODataPagingService.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                pageLinkData = new PageLinkData(requestInfo, pageData, pageLinkData.ResponseFormat, pageLinkData.ItemName, pageLinkData.NextLinkName);
+                nextLink = await GetNextPageLinkAsync(pageLinkData, cancellationToken);
+                if (nextLink != null) pageLinkData.RequestInformation.URI = nextLink;
+            }
+            else
+            {
+                nextLink = null;
+            }
+            response = await MergeJsonStreamsAsync(response, pageData, pageLinkData.ItemName, pageLinkData.NextLinkName, cancellationToken);
+        } while (nextLink != null);
+
+        return response ?? Stream.Null;
+    }
+
+    /// <summary>
+    /// Merges 2 streams of JSON on the property defined by <code>itemName</code>. The property should be a JSON array
+    /// </summary>
+    /// <param name="left">The first stream.</param>
+    /// <param name="right">The second stream.</param>
+    /// <param name="itemName">The name of the array property to merge on.</param>
+    /// <param name="nextLinkName">The name of the property containing the next link name.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    internal async Task<Stream?> MergeJsonStreamsAsync(Stream? left, Stream? right, string itemName = "value", string nextLinkName = "nextLink", CancellationToken cancellationToken = default)
+    {
+        if (left?.CanSeek == true) left?.Seek(0, SeekOrigin.Begin);
+        if (right?.CanSeek == true) right?.Seek(0, SeekOrigin.Begin);
+        if (left == null || right == null)
+        {
+            return left ?? right;
+        }
+
+        JsonNode? nodeLeft = null;
+        if (left != null)
+        {
+            nodeLeft = JsonNode.Parse(left);
+            if (left?.CanSeek == true) left?.Seek(0, SeekOrigin.Begin);
+        }
+        JsonNode? nodeRight = null;
+        if (right != null)
+        {
+            nodeRight = JsonNode.Parse(right);
+            if (right?.CanSeek == true) right?.Seek(0, SeekOrigin.Begin);
+        }
+
+        JsonArray? leftArray = null;
+        JsonArray? rightArray = null;
+        if (!string.IsNullOrWhiteSpace(itemName))
+        {
+            if (nodeLeft?[itemName] == null || nodeRight?[itemName] == null)
+            {
+                return left ?? right;
+            }
+
+            leftArray = nodeLeft[itemName]?.AsArray();
+            rightArray = nodeRight[itemName]?.AsArray();
+        }
+        else
+        {
+            leftArray = nodeLeft?.AsArray();
+            rightArray = nodeRight?.AsArray();
+        }
+
+
+        if (leftArray != null && rightArray != null)
+        {
+            var elements = rightArray.Where(i => i != null);
+            var item = elements.FirstOrDefault();
+            while (item != null)
+            {
+                rightArray.Remove(item);
+                leftArray.Add(item);
+                item = elements.FirstOrDefault();
+            }
```

Comment:
The item has to be removed from the right array first before being added to the right array. Otherwise, the code will throw an exception. Your version is what I tried first.